go-client/client/example_table_ops_test.go

```diff
@@ -34,15 +102,95 @@ func Example_basicQuery() {
 	// Note that Arrow records must eventually be released.
 	defer sampleRecord.Release()
 
-	fmt.Println("Data Before:")
-	fmt.Println(sampleRecord)
+	// Now we upload the record as a table on the server.
+	// We get back a TableHandle, which is a reference to a table on the server.
+	baseTable, err := cl.ImportTable(ctx, sampleRecord)
+	if err != nil {
+		fmt.Println("error when uploading table:", err.Error())
+		return "", err
+	}
+	// Table handles should be released when they are no longer needed.
+	defer baseTable.Release(ctx)
+
+	// Now, let's start doing table operations.
+	// Maybe I don't like companies whose names are too long or too short, so let's keep only the ones in the middle.
+	midStocks, err := baseTable.Where(ctx, "Ticker.length() == 3 || Ticker.length() == 4")
+	if err != nil {
+		fmt.Println("error when filtering table:", err.Error())
+		return "", err
+	}
+	defer midStocks.Release(ctx)
+
+	// We can also create completely new tables with the client too.
+	// Let's make a table whose columns are powers of ten.
+	powTenTable, err := cl.EmptyTable(ctx, 10)
+	if err != nil {
+		fmt.Println("error when creating an empty table:", err.Error())
+		return "", err
+	}
+	defer powTenTable.Release(ctx)
+	powTenTable, err = powTenTable.Update(ctx, "Magnitude = (int)pow(10, ii)")
+	if err != nil {
+		fmt.Println("error when updating a table:", err.Error())
+		return "", err
+	}
+	defer powTenTable.Release(ctx)
+
+	// What if I want to bin the companies according to the magnitude of the Volume column?
+	// We can perform an as-of join between two tables to produce another table.
+	magStocks, err := midStocks.
+		AsOfJoin(ctx, powTenTable, []string{"Volume = Magnitude"}, nil, client.MatchRuleLessThanEqual)
+	if err != nil {
+		fmt.Println("error when doing an as-of join:", err.Error())
+		return "", err
+	}
+	defer magStocks.Release(ctx)
+
+	// Now, if we want to see the data in each of our tables, we can take snapshots.
+	midRecord, err := midStocks.Snapshot(ctx)
+	if err != nil {
+		fmt.Println("error when snapshotting:", err.Error())
+		return "", err
+	}
+	defer midRecord.Release()
+	magRecord, err := magStocks.Snapshot(ctx)
+	if err != nil {
+		fmt.Println("error when snapshotting:", err.Error())
+		return "", err
+	}
+	defer magRecord.Release()
+
+	return fmt.Sprintf("Data Before:\n%s\nNew data:\n%s\n%s", sampleRecord, midRecord, magRecord), nil
+}
+
+// This function demonstrates how to use the query system.
+//
+// The query system allows you to build up an arbitrary number of table operations into a single object,
+// and then execute all of the table operations at once.
+// This simplifies error handling, is much more concise, and can be more efficient than doing the operations separately.
+func doQueryOps() (string, error) {
```

Comment:
"query system" also feels like weird nomenclature, because aren't the "normal" operations also doing queries.

Reply:
Maybe you want to call the two options Concise and Verbose or Sequential and Batch?